File: src/main/java/com/qtech/bubbles/effect/BubbleFreezeEffect.java
package com.qtech.bubbles.effect;

import com.qtech.bubbles.common.effect.StatusEffect;
import com.qtech.bubbles.common.effect.StatusEffectInstance;
import com.qtech.bubbles.entity.Entity;
import com.qtech.bubbles.entity.player.PlayerEntity;
import com.qtech.utilities.python.builtins.ValueError;

public class BubbleFreezeEffect extends StatusEffect {
    public BubbleFreezeEffect() throws ValueError {
        super();
    }

    @Override
    protected boolean canExecute(Entity entity, StatusEffectInstance statusEffectInstance) {
        return false;
    }

    @Override
    public void onStart(StatusEffectInstance statusEffectInstance, Entity entity) {
        if (entity instanceof PlayerEntity) {
            if (!entity.getGameType().isGlobalBubbleFreeze()) {
                entity.getGameType().setGlobalBubbleFreeze(true);
            }
        }
    }

    @Override
    public void onStop(Entity entity) {
        if (entity instanceof PlayerEntity) {
            if (entity.getGameType().isGlobalBubbleFreeze()) {
                entity.getGameType().setGlobalBubbleFreeze(false);
            }
        }
    }
}

File: src/main/kotlin/com/qtech/bubbles/effect/BubbleFreezeEffect.kt
package com.qtech.bubbles.effect

import com.qtech.bubbles.common.effect.Effect
import com.qtech.bubbles.common.effect.EffectInstance
import com.qtech.bubbles.entity.Entity
import com.qtech.bubbles.entity.player.PlayerEntity

class BubbleFreezeEffect : Effect() {
    override fun canExecute(entity: Entity, effectInstance: EffectInstance): Boolean {
        return false
    }

    override fun onStart(effectInstance: EffectInstance, entity: Entity) {
        if (entity is PlayerEntity) {
            if (!entity.gameMode.isGlobalBubbleFreeze) {
                entity.gameMode.isGlobalBubbleFreeze = true
            }
        }
    }

    override fun onStop(entity: Entity?) {
        if (entity is PlayerEntity) {
            if (entity.gameMode.isGlobalBubbleFreeze) {
                entity.gameMode.isGlobalBubbleFreeze = false
            }
        }
    }
}